src/test/java/com/artipie/npm/proxy/NpmProxyTest.java

```diff
@@ -0,0 +1,318 @@
+        );
+        MatcherAssert.assertThat("Metadata is null", metadata != null);
+        final JSONObject json = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE)
+            .parse(metadata, JSONObject.class);
+        MatcherAssert.assertThat(
+            json.getAsString("last-modified"),
+            new IsEqual<>(NpmProxyTest.DEF_LAST_MODIFIED)
+        );
+    }
+
+    @Test
+    public void cannotFindPackage(final VertxTestContext context) {
+        MatcherAssert.assertThat(
+            "Unexpected package found",
+            this.npm.getPackage("not-found").blockingGet() == null
+        );
+        context.completeNow();
+    }
+
+    @Test
+    public void getsAsset(final VertxTestContext context) throws ParseException {
+        final NpmAsset asset = this.npm.getAsset("asdas/-/asdas-1.0.0.tgz").blockingGet();
+        context.completeNow();
+        MatcherAssert.assertThat("Asset is null", asset != null);
+        MatcherAssert.assertThat(
+            new String(
+                new Concatenation(asset.dataPublisher()).single().blockingGet().array(),
+                StandardCharsets.UTF_8
+            ),
+            new IsEqual<>(NpmProxyTest.DEF_CONTENT)
+        );
+        MatcherAssert.assertThat(
+            asset.lastModified(),
+            new IsEqual<>(NpmProxyTest.DEF_LAST_MODIFIED)
+        );
+        MatcherAssert.assertThat(
+            asset.contentType(),
+            new IsEqual<>(NpmProxyTest.DEF_CONTENT_TYPE)
+        );
+        final BlockingStorage bstorage = new BlockingStorage(this.storage);
+        final String content = new String(
+            bstorage.value(new Key.From("asdas/-/asdas-1.0.0.tgz")),
+            StandardCharsets.UTF_8
+        );
+        MatcherAssert.assertThat(NpmProxyTest.DEF_CONTENT, new IsEqual<>(content));
+        final String metadata = new String(
+            bstorage.value(new Key.From("asdas/-/asdas-1.0.0.tgz.metadata")),
+            StandardCharsets.UTF_8
+        );
+        MatcherAssert.assertThat("Metadata is null", metadata != null);
+        final JSONObject json = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE)
+            .parse(metadata, JSONObject.class);
+        MatcherAssert.assertThat(
+            json.getAsString("last-modified"),
+            new IsEqual<>(NpmProxyTest.DEF_LAST_MODIFIED)
+        );
+        MatcherAssert.assertThat(
+            json.getAsString("content-type"),
+            new IsEqual<>(NpmProxyTest.DEF_CONTENT_TYPE)
+        );
+    }
+
+    @Test
+    public void cannotFindAsset(final VertxTestContext context) {
+        MatcherAssert.assertThat(
+            "Unexpected asset found",
+            this.npm.getAsset("not-found").blockingGet() == null
+        );
+        context.completeNow();
+    }
+
+    @BeforeEach
+    void setUp(final Vertx svertx, final VertxTestContext context)
+        throws IOException, InterruptedException {
+        prepareServer(svertx, context);
+        this.cvertx = Vertx.vertx();
```

Comment:
Why do you create another instance of Vert.x? Are not the first one is okay?

Reply:
Fixed. The problem was that HTTP Server stops asynchronously and I have an issue with testing service unavailable scenario. So I stop the whole vertx server instance. Now I have found close method with handler (I missed it initially) and I don't need the second vertx instance now